src/compiler/checker.ts

```diff
@@ -14558,9 +14558,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 constraint = getConstraintOfTypeParameter(typeVariable);
             }
             else {
-                const modifiersConstraint = getConstraintOfType(getModifiersTypeFromMappedType(type));
-                if (modifiersConstraint) {
-                    constraint = getApparentType(modifiersConstraint);
+                const modifiersType = getModifiersTypeFromMappedType(type);
+                if (modifiersType) {
+                    constraint = getApparentType(getConstraintOfType(modifiersType) || modifiersType);

```

Comment:
I'm not particularly fond of this - maybe there is a better way of doing this. This mainly tries to get the apparent type of the modifiers type when the modifiers type is a mapped type itself (like in `Mapped<Mapped2<Foo>>`). Mapped types don't return anything from `getConstraintOfType`.